src/bootloaders/syslinux.c

```diff
@@ -172,19 +172,17 @@ static bool syslinux_set_default_kernel(const BootManager *manager, const Kernel
                                 abort();
                         }
                 }
-                if (streq(k->path, kernel->path)) {
```

Comment:
Hm have to disagree, we still continue even without a default kernel, which may be NULL, and try to use it within that loop (regardless of its NULL or not).